### ISSUE TITLE
Generate cropped png images

### ISSUE THREAD

**Issue by diresu** [Abjad/abjad#1066]
Hi,

I would love to be able to generate cropped png images in different resolutions - but couldn't find any way to do so...

So I wonder how you think about adding some code to PersistenceManager.py in order to allow these things?

I appended a sketch of my idea...

Thanks for Abjad - and have a great day,

Dietrich Bollmann

---
```python
# An example:
import abjad
duration = abjad.Duration(1, 4)
notes = [abjad.Note(pitch, duration) for pitch in range(8)]
staff = abjad.Staff(notes)
abjad.show(staff)

# Now this could be rendered as png using the following:
abjad.persist(staff).as_png()

# How about allowing to provide the flags for lilypond in order to crop the image and/or to change the resolution?
abjad.persist(staff).as_png(flags='-dpreview')
abjad.persist(staff).as_png(flags='-dresolution=600')
abjad.persist(staff).as_png(flags='-dpreview -dresolution=600')

# This could also be done with '-dbackend=eps --png':
abjad.persist(staff).as_png(flags='-dbackend=eps --png')
abjad.persist(staff).as_png(flags='-dbackend=eps -dresolution=600 --png')

# It would be easy as well to add some named parameters to as_png():
abjad.persist(staff).as_png()
abjad.persist(staff).as_png(resolution=600)
abjad.persist(staff).as_png(preview=True)
abjad.persist(staff).as_png(preview=True, resolution=600)

```

Here the sketch of a patch:

```
@@ -254,6 +254,9 @@
         png_file_path=None,
         remove_ly=False,
         illustrate_function=None,
+        flags='--png',
+        preview=False,
+        resolution=False,
         **keywords
         ):
         """
@@ -318,11 +321,17 @@
             )
         shutil.copy(original_ly_file_path, temporary_ly_file_path)
 
+        # render lilypond flags
+        if preview:
+            flags='-dpreview'
+        if resolution and isinstance(resolution, int):
+            flags += ' -dresolution={}'.format(resolution)
+        
         timer = abjad.Timer()
         with timer:
             success = abjad.IOManager.run_lilypond(
                 temporary_ly_file_path,
-                flags='--png',
+                flags=flags,
                 )
         lilypond_rendering_time = timer.elapsed_time

```

**Comment by trevorbaca** [comment]
@diresu Added in development branch. Merging into mainline soon.

**Comment by trevorbaca** [comment]
Merged into main. Will be included in Abjad 3.1 release.
